### PR TITLE
Upgrade Blazor sample to rc2

### PR DESCRIPTION
**Pull request by benjaminpetit**
Related issue: #8035

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/orleans/pull/8040)

### CODE REVIEW

File: samples/Blazor/BlazorServer/Grains/TodoManagerGrain.cs [hunk 1]
```diff
@@ -29,6 +29,7 @@ public async Task UnregisterAsync(Guid itemKey)
     public Task<ImmutableArray<Guid>> GetAllAsync() =>
         Task.FromResult(ImmutableArray.CreateRange(_state.State.Items));
 
+    [GenerateSerializer]
     public class State
     {
         public HashSet<Guid> Items { get; set; } = new();
```
Comment (ReubenBond) [inline samples/Blazor/BlazorServer/Grains/TodoManagerGrain.cs:35]: Missing `[Id(x)]` attributes

File: samples/Blazor/BlazorServer/Models/WeatherInfo.cs [hunk 2]
```diff
@@ -1,8 +1,10 @@
+using Orleans;
 using Orleans.Concurrency;
 
 namespace BlazorServer.Models;
 
 [Immutable, Serializable]
```
Comment (ReubenBond) [inline samples/Blazor/BlazorServer/Models/WeatherInfo.cs:6]: `[Serializable]` is not required

File: samples/Blazor/BlazorWasm/BlazorWasm.Server/Grains/TodoGrain.cs [hunk 3]
```diff
@@ -44,8 +44,8 @@ await GrainFactory.GetGrain<ITodoManagerGrain>(item.OwnerKey)
             GrainType, GrainKey, item);
 
         // notify listeners - best effort only
-        GetStreamProvider("SMS")
-            .GetStream<TodoNotification>(item.OwnerKey, nameof(ITodoGrain))
+        this.GetStreamProvider("MemoryStreams")
+            .GetStream<TodoNotification>(StreamId.Create(nameof(ITodoGrain), item.OwnerKey))
```
Comment (ReubenBond) [inline samples/Blazor/BlazorWasm/BlazorWasm.Server/Grains/TodoGrain.cs:48]: Should we add a 2-parameter overload of `GetStream` accepting `(string ns, string key)`?

File: samples/Blazor/BlazorWasm/BlazorWasm.Server/Grains/TodoManagerGrain.cs [hunk 4]
```diff
@@ -28,6 +28,7 @@ public async Task UnregisterAsync(Guid itemKey)
     public Task<ImmutableArray<Guid>> GetAllAsync() =>
         Task.FromResult(ImmutableArray.CreateRange(_state.State.Items));
 
+    [GenerateSerializer]
     public class State
     {
         public HashSet<Guid> Items { get; set; } = new();
```
Comment (ReubenBond) [inline samples/Blazor/BlazorWasm/BlazorWasm.Server/Grains/TodoManagerGrain.cs:34]: `[Id(x)]`

**Review by ReubenBond — CHANGES_REQUESTED** [review]
See comments

File: samples/Blazor/BlazorWasm/BlazorWasm.Server/Grains/TodoGrain.cs (hunk 3 above)
Comment (benjaminpetit) [reply samples/Blazor/BlazorWasm/BlazorWasm.Server/Grains/TodoGrain.cs:48]: I think it's a good idea

**Pull request merged**